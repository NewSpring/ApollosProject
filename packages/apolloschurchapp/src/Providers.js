import React from 'react';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

import theme from './theme'; // where theme looks like:

const NewSpringProviders = (props) => (
  <Providers themeInput={theme}>
    <ClientProvider {...props} />
  </Providers>
);

export default NewSpringProviders;
