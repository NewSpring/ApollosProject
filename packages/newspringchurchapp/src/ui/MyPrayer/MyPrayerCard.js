import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure } from 'recompose';
// import { View } from 'react-native';
import moment from 'moment';

import {
  Card,
  withIsLoading,
  styled,
  CardContent,
  H5,
  UIText,
  PaddedView,
} from '@apollosproject/ui-kit';

const enhance = compose(
  withIsLoading,
  pure
);

const PrayerText = styled(() => ({
  textAlign: 'center',
}))(UIText);

const HeaderView = styled(() => ({
  paddingBottom: 0,
}))(PaddedView);

const MyPrayerCard = enhance(({ duration, text, ...otherProps }) => (
  <Card {...otherProps}>
    <HeaderView>
      <H5>{moment(duration).fromNow()}</H5>
    </HeaderView>
    <CardContent>
      <PrayerText>{text}</PrayerText>
    </CardContent>
  </Card>
));

MyPrayerCard.propTypes = {
  duration: PropTypes.string.isRequired,
  text: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default MyPrayerCard;
