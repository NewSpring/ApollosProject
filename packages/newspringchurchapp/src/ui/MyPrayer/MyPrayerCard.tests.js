import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import MyPrayerCard from './MyPrayerCard';

const prayer1 = {
  id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4787',
  firstName: 'Aaron',
  lastName: 'Attendee',
  text: 'This is a test to see if I have a prayer request.',
  enteredDateTime: '2017-04-08T11:21:53.053',
};
const prayer2 = {
  id: 'PrayerRequest:a043b2710b8d9c0d2f659ff90d00cf9d',
  firstName: 'Aaron',
  lastName: 'Attendee',
  text:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a tellus sit amet mi consectetur pharetra lacinia eget arcu. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Phasellus felis diam, sagittis ut.',
  enteredDateTime: '2017-04-10T13:50:22.753',
};
const prayer3 = {
  id: 'PrayerRequest:05c9c6351be882103edb1e350c77422b',
  firstName: 'Aaron',
  lastName: 'Attendee',
  text:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed pulvinar, elit nec lobortis aliquam, urna.',
  enteredDateTime: '2017-04-10T13:53:24.623',
};
const prayer4 = {
  id: 'PrayerRequest:1c627c20911791321f819125a65c3c9d',
  firstName: 'Aaron',
  lastName: 'Attendee',
  text:
    'Lorem Lorem Lorem Lorem Ipsum Ipsum Ipsum Dolor Dolor sit sit sit amet amet.',
  enteredDateTime: '2017-04-10T14:55:14.01',
};
const prayer5 = {
  id: 'PrayerRequest:66a4d75b02b447556e4e3806430a9946',
  firstName: 'Aaron',
  lastName: 'Attendee',
  text:
    'Aaron is the man with the plan who owns a can and drives to Japan which has a lot of sand.',
  enteredDateTime: '2017-04-10T15:14:54.013',
};

describe('the MyPrayerCard component', () => {
  it('renders Prayer1', () => {
    const tree = renderer.create(
      <Providers>
        <MyPrayerCard
          id={prayer1.id}
          text={prayer1.text}
          duration={prayer1.enteredDateTime}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders Prayer2', () => {
    const tree = renderer.create(
      <Providers>
        <MyPrayerCard
          id={prayer2.id}
          text={prayer2.text}
          duration={prayer2.enteredDateTime}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders Prayer3', () => {
    const tree = renderer.create(
      <Providers>
        <MyPrayerCard
          id={prayer3.id}
          text={prayer3.text}
          duration={prayer3.enteredDateTime}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders Prayer4', () => {
    const tree = renderer.create(
      <Providers>
        <MyPrayerCard
          id={prayer4.id}
          text={prayer4.text}
          duration={prayer4.enteredDateTime}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders Prayer5', () => {
    const tree = renderer.create(
      <Providers>
        <MyPrayerCard
          id={prayer5.id}
          text={prayer5.text}
          duration={prayer5.enteredDateTime}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
