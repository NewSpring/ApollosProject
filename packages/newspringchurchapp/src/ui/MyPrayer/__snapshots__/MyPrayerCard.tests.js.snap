// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MyPrayerCard component renders Prayer1 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      id="PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4787"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 0,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Medium",
                "fontSize": 14,
                "lineHeight": 20.16,
              }
            }
          >
            2 days ago
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "textAlign": "center",
              }
            }
          >
            This is a test to see if I have a prayer request.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the MyPrayerCard component renders Prayer2 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      id="PrayerRequest:a043b2710b8d9c0d2f659ff90d00cf9d"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 0,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Medium",
                "fontSize": 14,
                "lineHeight": 20.16,
              }
            }
          >
            3 hours ago
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "textAlign": "center",
              }
            }
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a tellus sit amet mi consectetur pharetra lacinia eget arcu. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Phasellus felis diam, sagittis ut.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the MyPrayerCard component renders Prayer3 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      id="PrayerRequest:05c9c6351be882103edb1e350c77422b"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 0,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Medium",
                "fontSize": 14,
                "lineHeight": 20.16,
              }
            }
          >
            3 hours ago
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "textAlign": "center",
              }
            }
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed pulvinar, elit nec lobortis aliquam, urna.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the MyPrayerCard component renders Prayer4 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      id="PrayerRequest:1c627c20911791321f819125a65c3c9d"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 0,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Medium",
                "fontSize": 14,
                "lineHeight": 20.16,
              }
            }
          >
            2 hours ago
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "textAlign": "center",
              }
            }
          >
            Lorem Lorem Lorem Lorem Ipsum Ipsum Ipsum Dolor Dolor sit sit sit amet amet.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the MyPrayerCard component renders Prayer5 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      id="PrayerRequest:66a4d75b02b447556e4e3806430a9946"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 0,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Medium",
                "fontSize": 14,
                "lineHeight": 20.16,
              }
            }
          >
            an hour ago
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "System",
                "fontSize": 14,
                "fontStyle": null,
                "fontWeight": null,
                "lineHeight": 20.16,
                "textAlign": "center",
              }
            }
          >
            Aaron is the man with the plan who owns a can and drives to Japan which has a lot of sand.
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
